pkg/tsdb/azuremonitor/metrics/url-builder.go

```diff
@@ -39,22 +42,22 @@ func (params *urlBuilder) buildResourceURIFromLegacyQuery() string {
 		provider,
 	}
 
-	for i, metricDefinition := range metricDefinitionArray {
-		urlArray = append(urlArray, metricDefinition, resourceNameArray[i])
+	for i, namespace := range metricNamespaceArray {
+		urlArray = append(urlArray, namespace, resourceNameArray[i])
 	}
 
 	resourceURI := strings.Join(urlArray, "/")
 	return resourceURI
 }
 
-// BuildMetricsURL checks the metric definition property to see which form of the url
+// BuildMetricsURL checks the metric properties to see which form of the url
 // should be returned
 func (params *urlBuilder) BuildMetricsURL() string {
 	resourceURI := params.ResourceURI
 
 	// Prior to Grafana 9, we had a legacy query object rather than a resourceURI, so we manually create the resource URI
 	if resourceURI == "" {
-		resourceURI = params.buildResourceURIFromLegacyQuery()
```

Comment:
removed "FromLegacyQueruy" since it will be the main query mechanism (not something legacy)